File: server-laravel-hybrid/app/Models/Scopes/PersonalProjectScope.php
<?php

namespace App\Models\Scopes;

use Illuminate\Database\Eloquent\Scope;

/**
 * PersonalProjectScope — user-ownership isolation for the personal group.
 *
 * Auto-discovered by Rhino's HasAutoScope trait. Constrains every query to the
 * authenticated user's own personal projects.
 */
class PersonalProjectScope implements Scope
{
    public function apply($builder, $model): void
    {
        if (app()->runningInConsole() && !app()->runningUnitTests()) {
            return;
        }

        $user = auth('sanctum')->user() ?? auth()->user();

        if ($user) {
            $builder->where('personal_projects.user_id', $user->getKey());
        }
    }
}

File: server-rails-hybrid/app/models/scopes/personal_project_scope.rb
# frozen_string_literal: true

module Scopes
  # PersonalProjectScope — user-owned isolation for the `personal` group.
  #
  # Auto-discovered by HasAutoScope. Constrains every PersonalProject query to
  # the current user's rows. With no request user (console/seeds) `user` is nil
  # and no filter is applied.
  class PersonalProjectScope < Rhino::ResourceScope
    def apply(relation)
      return relation unless user

      relation.where(user_id: user.id)
    end
  end
end
